File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    
    public static List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        if (strs.length == 0) {
            
            return new ArrayList<>();
        
        }
        
        try{
        for (String s : strs) {
            char[] ch = s.toCharArray();
            Arrays.sort(ch);
            String el = String.valueOf(ch);
            if (!map.containsKey(el)) map.put(el, new ArrayList<>());
            map.get(el).add(s);}}
        catch (Exception e){
            //nothing
        }
            
        return new ArrayList<>(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashMap = defaultdict(list)
        ans = []

        for i in strs:
            arranged =str(sorted(i))

            if arranged in hashMap:
                ans[hashMap[arranged]].append(i)
            else:
                ans.append([i])
                hashMap[arranged] = len(ans)-1
        return ans
                
